### ISSUE TITLE
Issue with page model

### ISSUE THREAD

**Issue by pedroborges** [getkirby-plugins/algolia-plugin#2]
I'm trying to use the auto index feature but it is not working due to the use of a page model.

The field `author` is just slug that refers to the author page. I use a page model method to get the author name:

```php
    // page model
    public function author() {
        $slug = 'authors/' . parent::author(); // pedro-borges

        return site()->pages()->find($slug)->fullName(); // Sr. Pedro Borges
    }
```

This is how I set the callback when registering the fields that should be indexed.

```php
    // inside algolia.templates config
    'author' => function($page) {
        return $page->author();
    },
```

When I enable `algolia.autoindex`, the page doesn't index at all. The Algolia API call detail shows an error 400 with the message: _No content in PUT request_. When I inspect the value of `$data` at the end of the `Kirby\Algolia::formatPage()` method, it show that the `author` field is `object(Field)`, not the string I expected.

The weird thing is that indexing manually by using `algolia()->index()` honours the page model returning the full author name.

 I spent the last couple of hours trying to solve this myself but couldn't :(. I would really appreciate any help.

**Comment by lukasbestle** [comment]
If you use a callback function, you must make sure that it always returns a value that can be serialized in JSON like a string, a number, a boolean or null.

Your page model returns a field, so you need to cast it to a string in your Algolia templates config.
You can also just omit the callback entirely so that `'author'` becomes a value in the array. The plugin will cast the field to a string for you then.

**Comment by pedroborges** [comment]
Thanks for the quick reply!

I had tried what you suggested before opening this issue and it didn't work too :(

When debugging I also found out that the page model never got called when auto indexing, so it's not a matter of casting the value to string. I think this has something to do with the page object passed to the hook coming straight from the panel (not having the model loaded), which is not the case when you use `algolia()->index()`. We have two different behaviours here.

I'm willing to dig deeper to see this fixed, just not sure how.

**Comment by lukasbestle** [comment]
Yes, that is definitely true. Autoindexing won't respect models and there is currently no easy way to make it do that. But if you omit the callback it should at least not throw an error/fail uploading to Algolia.

**Comment by pedroborges** [comment]
Omitting the callback works, just not as expected. It sends the raw field value, not the modified value processed on the page model.

**Comment by lukasbestle** [comment]
Yes, that is as expected given that the Panel does not load models. Closing this as the error is fixed and the feature request is noted in #3.
